pkg/controller/support/credentials/cloudauth/ecr/ecr_test.go

```diff
@@ -1,133 +1,198 @@
-//go:build integration
-// +build integration
-
 package ecr
 
 import (
 	"context"
 	"errors"
+	"fmt"
 	"reflect"
-	"strings"
 	"testing"
 
-	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/service/ecr"
-	"github.com/aws/aws-sdk-go-v2/service/ecr/types"
-	dockertypes "github.com/docker/docker/api/types"
-	"github.com/go-logr/logr"
-
-	"github.com/dominodatalab/hephaestus/pkg/controller/support/credentials/cloudauth"
+	ecrTypes "github.com/aws/aws-sdk-go-v2/service/ecr/types"
+	dockerTypes "github.com/docker/docker/api/types"
+	"github.com/go-logr/zapr"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest/observer"
+	"gotest.tools/v3/assert"
 )
 
-func TestRegister(t *testing.T) {
-	registry := &cloudauth.Registry{}
-
-	err := Register(logr.Discard(), registry)
-	if err != nil {
-		t.Error(err)
-	}
-}
-
-func TestPatternMatching(t *testing.T) {
-	testcases := []struct {
-		name  string
-		url   string
-		match bool
+func TestAuthenticate(t *testing.T) {
+	defaultCtx := context.Background()
+	observerCore, observedLogs := observer.New(zap.DebugLevel)
+	log := zapr.NewLogger(zap.New(observerCore))
+
+	var emptyToken *string
+	invalidB64 := "%"
+	invalidToken := "YWJk"
```

Comment:
these don't need to be declared at the top level. they are only used once.

Reply:
I have variables for those because `AuthorizationToken` takes a pointer to a string. Is there a better way to inline that?